.github/workflows/examples-esp32.yaml

```diff
@@ -122,6 +122,12 @@ jobs:
             - name: Build example Lighting App (external platform)
               run: scripts/examples/esp_example.sh lighting-app sdkconfig.ext_plat.defaults
 
+            - name: Build example Energy Gateway App
```

Comment:
could we separate out the "add more builds" and "fix compilation" parts?

I am fully ok with the compilation fix, however have concerns of adding more and more things to our CI.